tabby-core/src/services/profiles.service.ts

```diff
@@ -211,27 +349,192 @@ export class ProfilesService {
+    */
+
+    /**
+    * Return global defaults for a given profile provider
+    * Always return something, empty object if no defaults found
+    */
+    getProviderDefaults (provider: ProfileProvider<Profile>): any {
+        const defaults = this.config.store.profileDefaults
+        return defaults[provider.id] ?? {}
+    }
+
+    /**
+    * Set global defaults for a given profile provider
+    */
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    setProviderDefaults (provider: ProfileProvider<Profile>, pdefaults: any): void {
+        this.config.store.profileDefaults[provider.id] = pdefaults
+    }
+
+    /**
+    * Return defaults for a given profile
+    * Always return something, empty object if no defaults found
+    * arg: skipUserDefaults -> do not merge global provider defaults in ConfigProxy
+    */
+    getProfileDefaults (profile: PartialProfile<Profile>, skipGlobalDefaults = false, skipGroupDefaults = false): any[] {
         const provider = this.providerForProfile(profile)
-        const defaults = [
+
+        return [
             this.profileDefaults,
             provider?.configDefaults ?? {},
-            !provider || skipUserDefaults ? {} : this.config.store.profileDefaults[provider.id] ?? {},
-        ].reduce(configMerge, {})
-        return new ConfigProxy(profile, defaults) as unknown as T
+            provider && !skipGlobalDefaults ? this.getProviderDefaults(provider) : {},
+            provider && !skipGlobalDefaults && !skipGroupDefaults ? this.getProviderProfileGroupDefaults(profile.group ?? '', provider) : {},
+        ]
     }
 
-    async launchProfile (profile: PartialProfile<Profile>): Promise<void> {
-        await this.openNewTabForProfile(profile)
+    /*
+    * Methods used to interract with ProfileGroup
+    */
 
-        let recentProfiles: PartialProfile<Profile>[] = JSON.parse(window.localStorage['recentProfiles'] ?? '[]')
-        if (this.config.store.terminal.showRecentProfiles > 0) {
-            recentProfiles = recentProfiles.filter(x => x.group !== profile.group || x.name !== profile.name)
-            recentProfiles.unshift(profile)
-            recentProfiles = recentProfiles.slice(0, this.config.store.terminal.showRecentProfiles)
+    /**
+    * Synchronously return an Array of the existing ProfileGroups
+    * Does not return builtin groups
+    */
+    getSyncProfileGroups (): PartialProfileGroup<ProfileGroup>[] {
+        return deepClone(this.config.store.groups ?? [])
+    }
+
+    /**
+    * Return an Array of the existing ProfileGroups
+    * arg: includeProfiles (default: false) -> if false, does not fill up the profiles field of ProfileGroup
+    * arg: includeNonUserGroup (default: false) -> if false, does not add built-in and ungrouped groups
+    */
+    async getProfileGroups (includeProfiles = false, includeNonUserGroup = false): Promise<PartialProfileGroup<ProfileGroup>[]> {
```

Comment:
same here (options object)

Reply:
Resolved in 5e5c80832de2d2125a9e55d89321071365c32938